cluster/src/test/scala/org/apache/spark/sql/store/ColumnUpdateDeleteTest.scala

```diff
@@ -275,4 +302,86 @@ object ColumnUpdateDeleteTest extends Assertions {
     assert(res(0).getInt(0) === 73)
     assert(res(0).getString(1) === "addr73")
   }
+
+  def testSNAP1925(session: SnappySession): Unit = {
+    // reduced size to ensure both column table and row buffer have data
+    session.conf.set(Property.ColumnBatchSize.name, "10k")
+
+    val numElements = 50000
+
+    session.sql("drop table if exists order_details")
+    session.sql("create table order_details (OrderID int, ProductID int," +
+        "UnitPrice double, Quantity smallint, Discount double, tid int) " +
+        "using column options(partition_by 'OrderID', buckets '8')")
+
+    session.range(numElements).selectExpr("id", "id + 2", "1.0", "2", "rand()", "id + 1")
+        .write.insertInto("order_details")
+
+    session.sql("UPDATE order_details SET UnitPrice = UnitPrice * 1.1 WHERE tid = 6")
+
+    var result = session.sql("select UnitPrice, tid from order_details where tid <> 6").collect()
+    assert(result.length === numElements - 1)
+    assert(result.toSeq.filter(_.getDouble(0) != 1.0) === Seq.empty)
+
+    result = session.sql("select UnitPrice from order_details where tid = 6").collect()
+    assert(result.length === 1)
+    assert(result(0).getDouble(0) == 1.1)
+
+    session.sql("UPDATE order_details SET UnitPrice = UnitPrice * 1.1 WHERE tid <> 6")
+
+    result = session.sql("select UnitPrice from order_details where tid = 6").collect()
+    assert(result.length === 1)
+    assert(result(0).getDouble(0) == 1.1)
+    result = session.sql("select UnitPrice, tid from order_details where tid <> 6").collect()
+    assert(result.length === numElements - 1)
+    assert(result.toSeq.filter(_.getDouble(0) != 1.1) === Seq.empty)
+    result = session.sql("select UnitPrice, tid from order_details").collect()
+    assert(result.length === numElements)
+    assert(result.toSeq.filter(_.getDouble(0) != 1.1) === Seq.empty)
+
+
+    session.sql("UPDATE order_details SET UnitPrice = 1.1 WHERE tid <> 11")
+
+    result = session.sql("select UnitPrice from order_details where tid = 11").collect()
+    assert(result.length === 1)
+    assert(result(0).getDouble(0) == 1.1)
+    result = session.sql("select UnitPrice, tid from order_details where tid <> 6").collect()
+    assert(result.length === numElements - 1)
+    assert(result.toSeq.filter(_.getDouble(0) != 1.1) === Seq.empty)
+    result = session.sql("select UnitPrice, tid from order_details").collect()
+    assert(result.length === numElements)
+    assert(result.toSeq.filter(_.getDouble(0) != 1.1) === Seq.empty)
+
+    session.sql("drop table order_details")
+    session.conf.unset(Property.ColumnBatchSize.name)
+  }
+
+  def testSNAP1926(session: SnappySession): Unit = {
```

Comment:
The NPE in SNAP-1926 is not coming from code but from the Janino compilation. This test creates the exact same generated code that has been reported by Sonal to be failing with NPE in Janino. However it does not reproduce in this test so I am guessing that it is some race condition in the Janino compiler itself that manifests under stress conditions or an issue with older janino versions (though could not find any fix for this in its changelogs).

Since it is reported only in smart connector and upstream Spark uses an older version (3.0.0) compared to Snappy (3.0.6) so a bug in old Janino is the likely theory. I am hoping that with the changes in SNAP-1936 that significantly changes/reduces the generated code this will go away automatically, but if not we can recommend that Spark connector-mode users replace the janino 3.0.0 jar with the latest release 3.0.7 jar in their Spark product installation (and will ask Sonal to verify that the issue goes away with that).